llvm/test/CodeGen/AArch64/ptrauth-call.ll

```diff
@@ -188,6 +188,33 @@ define void @test_tailcall_omit_mov_x16_x16(ptr %objptr) #0 {
   ret void
 }
 
+define i32 @test_call_omit_extra_moves(ptr %objptr) #0 {
```

Comment:
As far as I see, there are two places where `emitPtrauthDiscriminator` is called with `/*MayUseAddrAsScratch=*/true`: `emitPtrauthBranch` and `emitInstruction` for `AUTH_TCRETURN`/`AUTH_TCRETURN_BTI`. Are both cases tested? It looks like that this new test case is designed for signed tail call which covers only one of two code paths, doesn't it?